### PR TITLE
Don't ignore dot directories or movies/episodes with sample in their name.

### LINKED ISSUE #3238: Jellyfin refuses to load series with the word Sample in Name
**Describe the bug**
Series with the word sample in the name get ignored and dont get imported in jellyfin

**System (please complete the following information):**
 - OS: Debian
 - Virtualization: Docker
 - Clients: Browser
 - Browser: Vivaldi 3.0
 - Jellyfin Version: 10.6.0 undefined
 - Playback: N/A
 - Installed Plugins: N/A
 - Reverse Proxy: None
 - Base URL:None
 - Networking: Host
 - Storage: Local

**To Reproduce**
Download an Anime like Shomin Sample
Add the serie to the libarry
Rename the anime to the full japanese name "俺がお嬢様学校に『庶民サンプル』としてゲッツされた件"
Watch the folder get made but nothing gets imported

**Expected behavior**
Serie gets imported normally

**Logs**
No log entries exists

**Screenshots**
https://imgur.com/a/GTxIeVR

**Additional context**
N/A

### LINKED ISSUE #3650: collections getting deleted after library scan
**Describe the bug**
I created some collections manually, after a library scan they were all gone.
i tried it multiple times

**System (please complete the following information):**
 - OS: Debian 10
 - Jellyfin Version: 10.6

**To Reproduce**

create collection manually, add movies manually, scan library, --> collections are gone

**Expected behavior**
collections not getting deleted after a library scan

**Logs**
[2020-07-21 21:52:46.493 +02:00] [DBG] [26] MediaBrowser.Controller.Entities.BaseItem: Removed item: /home/user/.config/Jellyfin/data/collections/Winter [boxset]
[2020-07-21 21:52:46.493 +02:00] [INF] [26] Emby.Server.Implementations.Library.LibraryManager: Removing item, Type: "BoxSet", Name: "Winter", Path: "/home/user/.config/Jellyfin/data/collections/Winter [boxset]", Id: 1f0ca47f-10bb-e7ec-ac24-be8d7e921de8

update:

- fresh jellyfin 10.6 install via swizzin on Debian Buster
- collections were created by tmdb boxset and manually
- all fine
- update library
- collections are gone in jellyfin, but are still in the collections folder

after a library scan jellyfin collections are empty, but the collections are still available on disk.
am i the only one experiencing this?

### LINKED ISSUE #3681: Regression on 10.6: Media files under a subtree of a hidden directory are excluded from the library scan
**Describe the bug**
On version 10.6.0, the contents of the folders that are added to the library are not scanned if the folder itself or any of its parent tree is a hidden directory.

Example: A path like `/opt/.library` or `/opt/.library/some-dir` content won't be scanned, as opposed to a directory like `/opt/library`.

This causes part of the library to disappear after upgrading to 10.6.

**System (please complete the following information):**
 - OS: Ubuntu 18.04
 - Virtualization: None
 - Clients: Web
 - Browser: Google Chrome
 - Jellyfin Version: 10.6.0
 - Playback: All types
 - Installed Plugins: None
 - Reverse Proxy: Nginx
 - Base URL: None
 - Networking: Host
 - Storage: Local

**To Reproduce**

1. Go to Admin - Dashboard - Libraries
2. Add a new library called A with type "Photos" with a folder `/opt/.a`
2. Add a new library called B with type "Photos" with a folder `/opt/.b/x`
4. Add a new library called C with type "Photos" with a folder `/opt/c`
5. Put some media files under all the directories mentioned above.
6. Trigger a scan on all libraries A, B and C.
7. Observe that only library C detects and adds the files.

(Follow the same steps on version 10.5.5 to see that all libraries are scanned correctly)

**Expected behavior**
The files under hidden directories are also scanned and added to the library.

### PR DESCRIPTION
**Pull request by oddstr13**
Use `.ignore` file to hide directory from library scan.

Also, please tell me we handle sample matching somewhere else? This is a mess.

**Changes**

Don't ignore dot directories or movies/episodes with sample in their name.

**Issues**

Fixes #3238
Fixes #3650 
Fixes #3681

### CODE REVIEW

File: Emby.Server.Implementations/Library/IgnorePatterns.cs [hunk 1]
```diff
@@ -18,7 +18,19 @@ public static class IgnorePatterns
         {
             "**/small.jpg",
             "**/albumart.jpg",
-            "**/*sample*",
+
+            // What is this reg ex you speak of?
```
Comment (cvium) [inline Emby.Server.Implementations/Library/IgnorePatterns.cs:22]: Funny, but useless comment. Explain what it does and/or why it's needed instead

File: Emby.Server.Implementations/Library/IgnorePatterns.cs [hunk 2]
```diff
@@ -64,10 +76,13 @@ public static class IgnorePatterns
             "**/.grab/**",
             "**/.grab",
 
-            // Unix hidden files and directories
-            "**/.*/**",
+            // Unix hidden files
             "**/.*",
```
Comment (crobibero) [inline Emby.Server.Implementations/Library/IgnorePatterns.cs:80]:
```suggestion
            "**/.*[!/]",
```

And add this test: `[InlineData("/media/music/Big K.R.I.T.", false)]` 
to fix #3635

Comment (oddstr13) [reply Emby.Server.Implementations/Library/IgnorePatterns.cs:80]: Well, that's annoying.
This would break dotfile matching.

Comment (oddstr13) [reply Emby.Server.Implementations/Library/IgnorePatterns.cs:80]: I fail to see how that's getting matched in the first place… It should only match paths who's filename starts with a dot…

Comment (crobibero) [reply Emby.Server.Implementations/Library/IgnorePatterns.cs:80]: > I fail to see how that's getting matched in the first place… It should only match paths who's filename starts with a dot…

That was my thought, but I brought the matching code out to a test project and that's the pattern that matched it

Comment (oddstr13) [reply Emby.Server.Implementations/Library/IgnorePatterns.cs:80]: I don't think we currently can fix this short of changing to regex or a different glob library.
I submitted an upstream issue; https://github.com/dazinator/DotNet.Glob/issues/78

**Comment by nekocentral** [comment]
I'm wondering would it be possible to also ignore the lost+found folder that gets created in a filesystem after a check?

**Comment by oddstr13** [comment]
> I'm wondering would it be possible to also ignore the lost+found folder that gets created in a filesystem after a check?

The pattern is already there, if it doesn't work then this particular function isn't the appropriate place to fix whatever issue you're having

**Comment by nekocentral** [comment]
I will do a pull later today to see if it's fixed for me, will report back that I you

**Pull request merged**